src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -269,9 +269,27 @@ static bool filterConversion(Conversion conversion)
                     ensureAllUnderlyingConversionsChecked(syntax, source, conversion, wasCompilerGenerated, destination, diagnostics);
                 }
 
+                var boundSource = BindToNaturalType(source, diagnostics);
+
+                if (!hasErrors && !isCast && conversion.Kind == ConversionKind.ImplicitPointer &&
```

Comment:
>if (!hasErrors && !isCast && conversion.Kind == ConversionKind.ImplicitPointer && [](http://example.com/codeflow?start=16&length=81)

I would put BindToNaturalType where it used to be and move this check above ```ensureAllUnderlyingConversionsChecked``` call (referring to the original source), where all top level conversions are validated.